kingpin/actors/packagecloud.py

```diff
@@ -0,0 +1,256 @@
+                            distro_version=package['distro_version'],
+                            filename=package['filename']
+                        ).http_delete()
+
+
+class Delete(PackagecloudBase):
+
+    """packagecloude Delete Base Object"""
+
+    all_options = {
+        'number_to_keep': (
+            int, 0,
+            'Keep at least this number of each package'),
+        'packages_to_delete': (
+            str, REQUIRED,
+            'Regex of packages to delete, e.g. package1|package2'),
+        'repo': (
+            str, REQUIRED,
+            'Which packagecloud repo to delete from'),
+    }
+
+    @gen.coroutine
+    def _execute(self):
+        """Deletes all packages that match the `packages_to_delete` pattern
+
+        raises: gen.Return()
+        """
+        yield self._delete()
+
+
+class DeleteByDate(PackagecloudBase):
+
+    """packagecloude DeleteByDate Base Object"""
+
+    all_options = {
+        'number_to_keep': (
+            int, 0,
+            'Keep at least this number of each package'),
+        'older_than': (
+            int, REQUIRED,
+            'Delete packages created before this number of seconds'),
+        'packages_to_delete': (
+            str, REQUIRED,
+            'Regex of packages to delete, e.g. package1|package2'),
+        'repo': (str, REQUIRED,
+                 'Which packagecloud repo to delete from')
+    }
+
+    @gen.coroutine
+    def _execute(self):
+        """Deletes all packages that match the `packages_to_delete` pattern and
+        where created before `older_than` number of seconds
+
+        raises: gen.Return()
+        """
+        yield self._delete(by_date=True)
+
+
+class WaitForPackage(PackagecloudBase):
+
+    """packagecloude WaitForPackage Base Object"""
+
+    all_options = {
+        'name': (
+            str, REQUIRED,
+            'Name of the package to search for'),
+        'version': (
+            str, '.*',
+            'Version of the package to search for'),
+        'repo': (str, REQUIRED,
+                 'Which packagecloud repo to search')
+    }
+
+    @gen.coroutine
+    def _execute(self):
+        """Searches for a given package until found or a timeout occurs"""
+        while True:
+            log.info('Searching for %s %s...' % (
+                self.option('name'), self.option('version')))
+
+            all_packages = yield self._packagecloud_client.packages(
+                token=TOKEN, account=ACCOUNT, repo=self.option('repo')
+            ).http_get()
+            log.debug('Found all packages: %s' % all_packages)
+
+            version_pattern = re.compile(self.option('version'))
+            found = False
+
+            for package in all_packages:
+                if package['name'] == self.option('name'):
+                    if version_pattern.match(package['version']):
+                        log.info('Found package %s' % package)
+                        found = True
```

Comment:
Skip the nested loop and the double break:

    matched_packages = [p for p in package
        if p['name'] == self.option('name') and version_pattern.match(p['version'])]

    if len(matched_packages) > 0:
        raise gen.Return()